src/test/testRunner.ts

```diff
@@ -41,14 +41,16 @@ async function main(): Promise<void> {
       'out',
       'userdata',
       'User',
-      'settings.json',
+      'settings.json'
     );
     fs.mkdirSync(path.dirname(settings_dst), { recursive: true });
     fs.copyFileSync(settings_src, settings_dst);
 
     // Install the latest released redhat.ansible extension
     const installLog = cp.execSync(
-      `"${cliPath}" ${cliArgs.join(' ')} --install-extension redhat.ansible --force`
+      `"${cliPath}" ${cliArgs.join(
```

Comment:
Seems like you include a number of unrelated formatting changes that shouldn't be bundled with functional changes because it makes it hard to find them in the diff.

Reply:
Same here as well !

Reply:
> I am quite happy to see it passing on CI. Still, when I downloaded the same change locally it failed to pass some of them.
> 
> ```
>   TEST EXTENSION IN LOCAL ENVIRONMENT
>     TEST FOR HOVER
> 
>       Hover for play keyworks
> 
>         ✔ should hover over `name` keyword
> 
>         ✔ should hover over `hosts` keyword
> 
>         ✔ should hover over `tasks` keyword
> 
>       Hover for task keyworks
> 
>         ✔ should hover over builtin module name
> 
>         1) should hover over collection module name
> 
>         ✔ should hover over task keyword
> 
>       Hover for module options and sub-options
> 
>         ✔ should hover over builtin module option
> 
>         2) should hover over collection module option (opt_1)
> 
>         3) should hover over collection module sub-option (opt_1 -> sub_opt_1)
> 
>         4) should hover over collection module sub-option (opt_1 -> sub_opt_2 -> sub_sub_opt_2)
> 
>         5) should hover over collection module sub-option (opt_1 -> sub_opt_2 -> sub_sub_opt_2 -> sub_sub_sub_opt_1)
> 
>   6 passing (6s)
>   5 failing
> 
>   1) TEST EXTENSION IN LOCAL ENVIRONMENT
>        TEST FOR HOVER
>          Hover for task keyworks
>            should hover over collection module name:
> 
>       AssertionError: expected 0 to equal 1
>       + expected - actual
> 
>       -0
>       +1
> 
>   	at testHover (out/client/test/helper.js:80:19)
>   	at processTicksAndRejections (internal/process/task_queues.js:93:5)
>   	at async Context.<anonymous> (out/client/test/testScripts/hover/testHoverLocal.test.js:70:17)
> 
> 
>   2) TEST EXTENSION IN LOCAL ENVIRONMENT
>        TEST FOR HOVER
>          Hover for module options and sub-options
>            should hover over collection module option (opt_1):
> 
>       AssertionError: expected 0 to equal 1
>       + expected - actual
> 
>       -0
>       +1
> 
>   	at testHover (out/client/test/helper.js:80:19)
>   	at processTicksAndRejections (internal/process/task_queues.js:93:5)
>   	at async Context.<anonymous> (out/client/test/testScripts/hover/testHoverLocal.test.js:97:17)
> 
> 
>   3) TEST EXTENSION IN LOCAL ENVIRONMENT
>        TEST FOR HOVER
>          Hover for module options and sub-options
>            should hover over collection module sub-option (opt_1 -> sub_opt_1):
> 
>       AssertionError: expected 0 to equal 1
>       + expected - actual
> 
>       -0
>       +1
> 
>   	at testHover (out/client/test/helper.js:80:19)
>   	at processTicksAndRejections (internal/process/task_queues.js:93:5)
>   	at async Context.<anonymous> (out/client/test/testScripts/hover/testHoverLocal.test.js:104:17)
> 
> 
>   4) TEST EXTENSION IN LOCAL ENVIRONMENT
>        TEST FOR HOVER
>          Hover for module options and sub-options
>            should hover over collection module sub-option (opt_1 -> sub_opt_2 -> sub_sub_opt_2):
> 
>       AssertionError: expected 0 to equal 1
>       + expected - actual
> 
>       -0
>       +1
> 
>   	at testHover (out/client/test/helper.js:80:19)
>   	at processTicksAndRejections (internal/process/task_queues.js:93:5)
>   	at async Context.<anonymous> (out/client/test/testScripts/hover/testHoverLocal.test.js:111:17)
> 
> 
>   5) TEST EXTENSION IN LOCAL ENVIRONMENT
>        TEST FOR HOVER
>          Hover for module options and sub-options
>            should hover over collection module sub-option (opt_1 -> sub_opt_2 -> sub_sub_opt_2 -> sub_sub_sub_opt_1):
> 
>       AssertionError: expected 0 to equal 1
>       + expected - actual
> 
>       -0
>       +1
> 
>   	at testHover (out/client/test/helper.js:80:19)
>   	at processTicksAndRejections (internal/process/task_queues.js:93:5)
>   	at async Context.<anonymous> (out/client/test/testScripts/hover/testHoverLocal.test.js:118:17)
> 
> 
> [mochawesome] Report JSON saved to /Users/ssbarnea/c/a/vscode-ansible/out/e2eTestReport/e2e_test_report.json
> 
> [mochawesome] Report HTML saved to /Users/ssbarnea/c/a/vscode-ansible/out/e2eTestReport/e2e_test_report.html
> 
> 
> Error: 1 tests failed.
> 
> [main 2022-01-12T10:54:01.164Z] Waiting for extension host with pid 98670 to exit.
> 
> [main 2022-01-12T10:54:01.164Z] Extension host with pid 98670 exited with code: 1, signal: null.
> 
> Exit code:   1
> Done
> 
> Failed to run tests due to exception!
> Failed
> ERROR    Received exit code 1 from: (['npm', 'run', 'test:e2e'],)
> mk test:e2e  17.59s user 4.05s system 119% cpu 18.059 total
> FAIL: 1
> ```

I suspect that npm is not setting up the dummy collections path for you in the environment !

Reply:
If it's prettier — submit these things separately. I'd prefer to be able to find things in the PR.

Reply:
Also, you can always `git commit -p` to select what to include.